### PR TITLE
Optionally allow flags to be specified on config/script

### PR DESCRIPTION
**Pull request by thecodefactory**
Related to https://github.com/libbitcoin/libbitcoin-explorer/issues/541

### CODE REVIEW

File: include/bitcoin/bitcoin/config/script.hpp [hunk 1]
```diff
@@ -83,6 +83,13 @@ class BC_API script
      */
     const std::string to_string() const;
 
+    /**
+     * Return a pretty-printed copy of the script.
+     * @param[in]  flags  The rule_fork flags to use.
+     * @return  A mnemonic-printed copy of the internal script.
+     */
+    const std::string to_string(uint32_t) const;
```
Comment (evoskuil) [inline include/bitcoin/bitcoin/config/script.hpp]: missing parameter name. why not use a default parameter (machine::rule_fork::all_rules) on a single method?

Comment (thecodefactory) [reply include/bitcoin/bitcoin/config/script.hpp]: Of course, that's better :+1:

Updated.

File: include/bitcoin/bitcoin/config/script.hpp [hunk 2]
```diff
@@ -79,9 +79,10 @@ class BC_API script
 
     /**
      * Return a pretty-printed copy of the script.
-     * @return  A mnemonic-printed copy of the internal script.
+     * @param[in]  flags  The rule fork flags to use.
+     * @return            A mnemonic-printed copy of the internal script.
      */
-    const std::string to_string() const;
+    const std::string to_string(uint32_t flags = machine::rule_fork::all_rules) const;
```
[the remark this answers is not in the document]
Comment (evoskuil) [reply include/bitcoin/bitcoin/config/script.hpp]:
```cpp
    const std::string to_string(
        uint32_t flags=machine::rule_fork::all_rules) const;
```
note spaces removed.

[the remark this answers is not in the document]
Comment (evoskuil) [reply include/bitcoin/bitcoin/config/script.hpp]: Also, there is no reason for const on the `const std::string` or `const bc::data_chunk` return types. Need to review other config classes for this.

[the remark this answers is not in the document]
Comment (thecodefactory) [reply include/bitcoin/bitcoin/config/script.hpp]: :+1: Thanks,  updated. Removing the unnecessary const return allows it to not be wrapped.

**Pull request merged**